### ISSUE TITLE
'decomp' test does not thoroughly check the return code of 'bfbcomp.csh'

### ISSUE THREAD

**Issue by phil-blain** [CICE-Consortium/CICE#524]
While investigating a failing 'decomp' test in a base_suite run from @JFLemieux73 for an upcoming PR, I noticed that the error-checking in the 'decomp' test script could use some hardening. Here is the part of the script that runs the `bfbcompare.csh` script:

https://github.com/CICE-Consortium/CICE/blob/0ac4cd9cc2934742d04187527d9ba258d3360cfe/configuration/scripts/tests/test_decomp.script#L74-L84

We just check if the script returns "0", and if not mark the test as FAIL with the message "bfbcomp and test dataset are different". 

However, we should be checking the return code of the  '`bfbcompare.csh`' script more thoroughly, so that the reason for the failure is not mis-diagnosed, since the script can return different failure codes in case of errors:

https://github.com/CICE-Consortium/CICE/blob/0ac4cd9cc2934742d04187527d9ba258d3360cfe/configuration/scripts/tests/comparebfb.csh#L13-L17

(as is done in [`baseline.script `](https://github.com/CICE-Consortium/CICE/blob/master/configuration/scripts/tests/baseline.script)).

@apcraig

**Comment by phil-blain** [comment]
I also realized that the decomp test is not re-runnable, because the `restart.${decomp}` folders are not cleaned up at the beginning of the test script. 

In fact this was why the test was failing in the first place; JF had mistakenly launched the suite several times, and the third time the  `mv -f` step failed because the destination was not empty (a little hard to explain but you'll see what I mean if you relaunch this test several times)...

**Comment by apcraig** [comment]
For the major tests (restart, smoke, etc), I have tried to build tests that were rerunable.  That is not always easy and not as easy with the decomp test.   I always though rerunable tests were a feature, not a requirement, but maybe we should change that.  

My general strategy is not to rerun.  Generally, I'll run a test suite and then if something fails, I'll setup a standalone case that duplicates the error.  I've found individual cases are easier to debug than a test, any test.  Once the error is fixed, I then rerun the test from scratch to confirm it passes.  We could get rid of the decomp test and change it to 12 separate tests and then the problem would be fixed.  There are pluses and minuses to different strategies.

Having said all that, I'll try to have a look at the rerunability issues with the decomp test and see if I can fix that.
